### PR TITLE
rtika blog 

### PR DESCRIPTION
**Pull request by predict-r**
try 2

### CODE REVIEW

**Comment by predict-r** [comment]
@stefaniebutland  Thanks, I added the yaml. 

Hmm, the "deploy/netlify' banner appeared for a second then disappeared along with its link to a preview.

Am I still doing something wrong?

**Comment by stefaniebutland** [comment]
See `Show all checks` in this screenshot image on the right. 

screen shot 2018-03-27 at 11 13 33 pm

Click that to see what looks like the image below with  `deploy/netlify` and `Details` on the right

screen shot 2018-03-27 at 11 14 36 pm

Click `Details` to get to the preview of our site. Click `Blog` and you should see this, the preview with your post

screen shot 2018-03-27 at 11 17 46 pm

**Comment by stefaniebutland** [comment]
Tentative date for your post is Tues Apr 17. I'll review your draft in detail next week.

In the meantime, 
- I notice the babel fish didn't come out right. I wonder if it's because line breaks don't render properly.
- you can format references as in this post: https://ropensci.org/blog/2017/12/05/rperseus/ ; https://raw.githubusercontent.com/ropensci/roweb2/master/content/blog/2017-12-05-rperseus.md

**Comment by predict-r** [comment]
Thank you so much for explaining it . I’ll try and fix the formatting after sleeping

**Comment by stefaniebutland** [comment]
@predict-r since tentative date for your post is Tues Apr 17, you can make any changes up to Tues Apr 10th. Let me know when you're finished/happy with it and I'll review then.

Cheers!

**Comment by stefaniebutland** [comment]
@predict-r Your post is scheduled to publish in one week on Tues Apr 24. 
Here is the preview: https://deploy-preview-166--ropensci.netlify.com/blog/2018/04/17/rtika-introduction/

Please update the date in YAML, make any final edits so I can do a final review. 
Note that the headings still look pretty big so best to reduce them.

Happy to answer any questions.

**Comment by predict-r** [comment]
It has smaller headings now and the date is updated. I'll be traveling until next Wed and will check emails for updates...off to the NYC text conference!!

File: content/blog/2018-04-03-rtika-introduction.md [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+
+``` r
+last_file <- length(batch)
+tryCatch(antiword(batch[last_file]), error = function(x){x})
+#> <simpleError: System call to 'antiword' failed (1): /Users/sasha/leginfo/2017/BILL_ANALYSIS_TBL_9999.lob is not a Word Document.
+```
+
+Fortunately, I remembered Apache Tika. Five years earlier, Tika helped parse the Internet Archive, and handled whatever format I threw at it. Back then, I put together a R package for myself and a few colleagues. It was outdated.
+
+I downloaded Tika and made a R script. Tika did its magic. It scanned the headers for the "Magic Bytes" [^4] and parsed the files appropriately:
```
Comment (stefaniebutland) [inline content/blog/2018-04-03-rtika-introduction.md]: *an* R script ?

File: content/blog/2018-04-03-rtika-introduction.md [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+timing[3]/2000
+#>  elapsed 
+#> 0.006245
+```
+
+For this batch, the efficiency compared favorably to `antiword`, even with the overhead of loading Tika. I estimate that starting Tika, loading the Java parsers each time, loading the file list from R, and reading the files back into an R object took a few extra seconds. The reduced time effort processing the entire batch led me to think about the broader applications of Tika. This was too good not to share, but I was apprehensive about maintaining a package over many years. The rOpenSci organization was ready to help.
+
+#### Lessons Learned
+
+I never distributed a package before on repositories such as CRAN or Github, and the rOpenSci group was the right place to learn how. The reviewers used a transparent on-boarding process and taught about good documentation and coding style. They were helping create a maintainable package by following certain standards. If I stopped maintaining `rtika`, others could use their knowledge of the same standards to take over. The vast majority of time was spent on documenting the code, the introductory vignette, and continuous testing to integrate new code.
```
Comment (stefaniebutland) [inline content/blog/2018-04-03-rtika-introduction.md]: onboarding, no hyphen

Comment (stefaniebutland) [reply content/blog/2018-04-03-rtika-introduction.md]: consider linking to http://onboarding.ropensci.org/ or https://github.com/ropensci/onboarding so others can see what it's about (I've asked in #onboarding channel which is link is preferable from blog post)

Comment (stefaniebutland) [reply content/blog/2018-04-03-rtika-introduction.md]: In "The reviewers used a transparent on-boarding process and taught about good documentation and coding style." you might link to the open review thread: https://github.com/ropensci/onboarding/issues/191

File: content/blog/2018-04-03-rtika-introduction.md [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+
+This worked. R sends Tika a signal to execute code using an old-fashioned command line call, telling Tika to parse a particular batch of files. R waits. Eventually, Tika sends the signal of its completion, and R can then return with results as a character vector. Surprisingly, this process may be a good option for containerized applications running Docker. In the example later in this blog post, a similar technique is used to connect to a Docker container in a few lines of code.
+
+Communication with Tika went smoothly, but after one issue with `base::system2()` was identified. That was terminating Tika's long running process. Switching to `sys::exec_wait()` or `processx::run()` solved the issue.
```
Comment (stefaniebutland) [inline content/blog/2018-04-03-rtika-introduction.md]: First two sentences here not clear - something missing?

File: content/blog/2018-04-03-rtika-introduction.md [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+
+##### The R User Interface
+
+Many in the R community make use of `magrittr` style pipelines, so those needed to work well. The Tidy Tools Manifesto makes piping a central tenant [^6], which makes code easier to read and maintain.
```
Comment (stefaniebutland) [inline content/blog/2018-04-03-rtika-introduction.md]: "central tenant" should be "central tenet"

File: content/blog/2018-04-03-rtika-introduction.md [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+
+When writing `rtika`, I created two distinct styles of user interface. The first was a lightweight R wrapper around the Tika command line, called `tika()`. The parameters and abbreviations there mirrored the Apache Tika tool. The other functions are in the common R style found in tidy tools, and run `tika()` with certain presets (e.g. `tika_html()` outputs to 'html' ). For R users, these should be more intuitive and self-documenting.
+
+##### Responding to Reviewers
+
+During the review process, I appreciated David Gohel's [^7] attention to technical details, and that Julia Silge [^8] and Noam Ross [^9] pushed me to make the documentation more focused. I ended up writing about each of the major functions in a vignette, one by one, in a methodical manner. While writing, I learned to understand Tika better.
```
Comment (stefaniebutland) [inline content/blog/2018-04-03-rtika-introduction.md]: This is awesome "I ended up writing about each of the major functions in a vignette, one by one, in a methodical manner. While writing, I learned to understand Tika better."

File: content/blog/2018-04-03-rtika-introduction.md [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+
+#### Tika in Context: Parsing the Internet Archive
+
+The first archive I parsed with Tika was a website retrieved from the Wayback Machine [^10], a treasure trove of historical files. Maintained by the Internet Archive, their crawler downloads sites over decades. When a site is crawled consistently, longitudinal analyses are possible. For example, federal agency websites often change when an administrations change, so the Internet Archive group and academic partners have increased the consistency of crawling there. In 2016, they archived over 200 terabytes of documents to include, among other things, over 40 million `pdf` files [^11]. I consider these government documents to be in the public domain, even if an administration hides or removes them.
```
Comment (stefaniebutland) [inline content/blog/2018-04-03-rtika-introduction.md]: "when an administrations change" needs a fix

File: content/blog/2018-04-03-rtika-introduction.md (hunk 2 above)
Comment (stefaniebutland) [reply content/blog/2018-04-03-rtika-introduction.md]: confirmed: please link to https://github.com/ropensci/onboarding

**Comment by stefaniebutland** [comment]
@predict-r Let me know when you've made the edits so I can publish

**Comment by predict-r** [comment]
Tomorrow I’ll be back from traveling and will review the changes on my laptop as opposed to this iPad. Thank you !

**Comment by stefaniebutland** [comment]
Sounds good. If it will be ready after ~noon Pacific time then please change the date to 2018-04-25 for filename, YAML date and any images

**Comment by stefaniebutland** [comment]
@predict-r Hope you're back safe and sound. 

Please make your final edits including date changes and tag me here so I can post asap today. We're publishing another post Thursday and I don't want them to detract attention from each other.

After edits are done:
> For tweets from rOpenSci about this post, do you have any lines from the post or specific things you would like to convey?

**Comment by predict-r** [comment]
I'm back and trying to merge your changes and suggestions now. Should be done in an hour.

**Comment by predict-r** [comment]
@stefaniebutland Your edits were merged into the file and committed. Its ready to go, as far as I'm concerned!

Today's date was added to the file name and yaml. 

The first paragraph of 'Connecting to Tika' was cleaned up, and hopefully it is clearer now and without added typos. 

I added a link to the rtika github page at the very end of the article. Let me know if that should be put elsewhere.

**Comment by predict-r** [comment]
> For tweets from rOpenSci about this post, do you have any lines from the post or specific things you would like to convey?

How about this:

"It currently handles text or metadata extraction from over one thousand digital formats"

**Comment by stefaniebutland** [comment]
@predict-r In the author spot, do you want to link to something other than the package? e.g. https://twitter.com/goodmansasha or a website?

**Comment by predict-r** [comment]
Yes! Please link to the twitter page: https://twitter.com/goodmansasha

**Comment by predict-r** [comment]
Thanks!! I see it is up on the site now.

**Pull request merged**